src/test-project/src/test/java/com/redhat/devtools/intellij/commonUiTestLibrary/fixturesTest/mainIdeWindow/toolWindowsPane/buildToolPane/GradleBuildToolPaneTest.java

```diff
@@ -0,0 +1,86 @@
+import com.redhat.devtools.intellij.commonUiTestLibrary.fixtures.mainIdeWindow.toolWindowsPane.ToolWindowsPane;
+import com.redhat.devtools.intellij.commonUiTestLibrary.fixtures.mainIdeWindow.toolWindowsPane.buildToolPane.GradleBuildToolPane;
+import com.redhat.devtools.intellij.commonUiTestLibrary.utils.project.CreateCloseUtils;
+import com.redhat.devtools.intellij.commonUiTestLibrary.utils.testExtension.ScreenshotAfterTestFailExtension;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.time.Duration;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Gradle Build Tool Pane test
+ *
+ * @author zcervink@redhat.com
+ */
+@ExtendWith(ScreenshotAfterTestFailExtension.class)
+class GradleBuildToolPaneTest extends LibraryTestBase {
+    private static final String projectName = "gradle_build_tool_pane_java_project";
+    private static ToolWindowsPane toolWindowsPane;
+    private static GradleBuildToolPane gradleBuildToolPane;
+
+    @BeforeAll
+    public static void prepareProject() {
+        CreateCloseUtils.createNewProject(remoteRobot, projectName, CreateCloseUtils.NewProjectType.GRADLE);
+        toolWindowsPane = remoteRobot.find(ToolWindowsPane.class, Duration.ofSeconds(10));
+        toolWindowsPane.openGradleBuildToolPane();
+        gradleBuildToolPane = toolWindowsPane.find(GradleBuildToolPane.class, Duration.ofSeconds(10));
+    }
+
+    @AfterAll
+    public static void closeCurrentProject() {
+        CreateCloseUtils.closeProject(remoteRobot);
+    }
+
+    @Test
+    public void buildProjectTest() {
+        gradleBuildToolPane.buildProject();
+        boolean isBuildSuccessful = toolWindowsPane.find(BuildView.class, Duration.ofSeconds(10)).isBuildSuccessful();
+        assertTrue(isBuildSuccessful, "The build should be successful but is not.");
+    }
+
+    @Test
+    public void reloadAllGradleProjects() {
+        gradleBuildToolPane.reloadAllGradleProjects();
+    }
+
+    @Test
+    public void expandAll() {
+        gradleBuildToolPane.collapseAll();
+        int itemsCountBeforeExpanding = gradleTaskTree().collectRows().size();
+        gradleBuildToolPane.expandAll();
+        int itemsCountAfterExpanding = gradleTaskTree().collectRows().size();
+        assertTrue(itemsCountAfterExpanding > itemsCountBeforeExpanding, "The 'Expand All' operation was unsuccessful.");
+    }
+
+    @Test
+    public void collapseAll() {
+        gradleTaskTree().expandAll();
+        int itemsCountBeforeCollapsing = gradleTaskTree().collectRows().size();
+        gradleBuildToolPane.collapseAll();
+        int itemsCountAfterCollapsing = gradleTaskTree().collectRows().size();
+        assertTrue(itemsCountAfterCollapsing < itemsCountBeforeCollapsing, "The 'Collapse All' operation was unsuccessful.");
+    }
+
+    private JTreeFixture gradleTaskTree() {
```

Comment:
It is already defined and visible in GradleBuildToolPane class.

Reply:
Yes I have forgotten about this when I made the implementation in GradleBuildToolPane public. Fixed.